li-utils/src/main/java/com/linkedin/common/urn/DashboardUrn.java

```diff
@@ -0,0 +1,57 @@
+package com.linkedin.common.urn;
+
+import com.linkedin.data.template.Custom;
+import com.linkedin.data.template.DirectCoercer;
+import com.linkedin.data.template.TemplateOutputCastException;
+import java.net.URISyntaxException;
+
+
+public final class DashboardUrn extends Urn {
+
+  public static final String ENTITY_TYPE = "dashboard";
+
+  private static final String CONTENT_FORMAT = "(%s,%s)";
+
+  private final String dashboardToolEntity;
+
+  private final String dashboardIdEntity;
+
+  public DashboardUrn(String dashboardTool, String dashboardId) {
+    super(ENTITY_TYPE, String.format(CONTENT_FORMAT, dashboardTool, dashboardId));
+    this.dashboardToolEntity = dashboardTool;
+    this.dashboardIdEntity = dashboardId;
+  }
+
+  public String getDashboardToolEntity() {
+    return dashboardToolEntity;
+  }
+
+  public String getDashboardIdEntity() {
+    return dashboardIdEntity;
+  }
+
+  public static DashboardUrn createFromString(String rawUrn) throws URISyntaxException {
```

Comment:
shouldn't we also check that ENTITY_TYPE is the same?

Reply:
Good catch. That's a bug, I guess we need to fix that for all other urns. But, that will be a separate PR.